netty/src/main/java/io/grpc/transport/netty/BufferingHttp2ConnectionEncoder.java

```diff
@@ -67,6 +69,44 @@
  * in replacement for {@link io.netty.handler.codec.http2.DefaultHttp2ConnectionEncoder}.
  */
 class BufferingHttp2ConnectionEncoder extends DecoratingHttp2ConnectionEncoder {
+
+  /**
+   * Thrown if buffered streams are terminated due to this encoder being closed.
+   */
+  public static final class ChannelClosedException extends Http2Exception {
+    public ChannelClosedException() {
+      super(Http2Error.STREAM_CLOSED, "Connection closed");
```

Comment:
STREAM_CLOSED is for more of a protocol error. It isn't applicable here.

REFUSED_STREAM seems more appropriate.